app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsEffectHandler.kt

```diff
@@ -54,16 +52,10 @@ class PatientSearchResultsEffectHandler @AssistedInject constructor(
     return ObservableTransformer { effects ->
       effects
           .observeOn(schedulers.io())
-          .switchMap {
-            userSession
-                .loggedInUser()
-                .extractIfPresent()
-                .switchMap(facilityRepository::currentFacility)
-                .take(1)
-                .observeOn(schedulers.ui())
-                .doOnNext(uiActions::openPatientEntryScreen)
-                .switchMap { Observable.empty<PatientSearchResultsEvent>() }
-          }
+          .map { currentFacility.get() }
+          .observeOn(schedulers.ui())
```

Comment:
Should we send a event to open patient entry screen instead of doing this?

Reply:
Not really, this is only been done this way because we need to switch threads separately, otherwise it would be a consumer.

An improvement I have planned is moving the thread switching for ui actions outside the effect handler, so we will not need to manually switch threads anymore.